Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -393,7 +430,7 @@ open class BasicFormat: SyntaxRewriter {
     trailingTrivia = trailingTrivia.indented(indentation: trailingTriviaIndentation, isOnNewline: false)
 
     leadingTrivia = leadingTrivia.trimmingTrailingWhitespaceBeforeNewline(isBeforeNewline: false)
-    trailingTrivia = trailingTrivia.trimmingTrailingWhitespaceBeforeNewline(isBeforeNewline: nextTokenWillStartWithNewline)
+    trailingTrivia = trailingTrivia.trimmingTrailingWhitespaceBeforeNewline(isBeforeNewline: nextTokenWillStartWithNewline || trailingTrivia.startsWithNewline)
```

Comment:
If we don't do it then 
```swift
      #"""
      """
      \(gen(xx) { (x) in
          return """
          case
      """
      })
      """
      """#
```

becomes 

```swift
      #"""
      """
      \(gen(xx) { (x) in
          return """
              case
      """
      })
      """
      """#
```

What I can see it is because it's moving spacing in front of the string segment that contains the `case`